packages/@aws-cdk/aws-ec2/lib/instance.ts

```diff
@@ -190,6 +213,22 @@ export interface InstanceProps {
    * @default - no association
    */
   readonly privateIpAddress?: string
+
+  /**
+   * Apply the given CloudFormation Init configuration to the instance at startup
+   *
+   * @default - no CloudFormation init
+   */
+  readonly init?: CloudFormationInit;
+
+  /**
+   * Use the given options for applying CloudFormation Init
+   *
+   * Describes the configsets to use and the timeout to wait
+   *
+   * @default - default options
+   */
+  readonly initOptions?: ApplyCloudFormationInitOptions;
```

Comment:
This could be made a parameter when `CloudFormationInit` is initialized via its factory methods. This would eliminate the need for the validation added later and seems like a better place. 

If you go down this path, `ApplyCloudFormationInitOptions` will become `CloudFormationInitOptions`.

Reply:
The context -- missing from this diff -- is that the options for ASGs are slightly different than the options for instances. Right now, that comes down to the modeling of timeouts/signals, because there are a broader set of options and a different mental model for that for a single instance vs a fleet of them. We could consolidate into one, and just document that the timeout-related option(s) are for instances only, but I worry the two might diverge more. Thoughts?

Reply:
I see. I'm not opposed to what you already have here.